### DISCUSSION TITLE
Allow to set API Keys permissions

### DISCUSSION THREAD

**Discussion by darekdesu** [twentyhq/twenty#5480] [Ideas]
## Scope & Context

Created API key has too much permissions (all of them :)) and can negatively affect security in case of its leakage. It also limits ability to share API token with any kind of 3rd party data providers.

## Current behavior

Cannot set the scope for created API key

## Expected behavior

Ability to set the scope for created API keys like on e.g. [on the Github](https://github.com/settings/tokens/new) itself where one could set `read,write,delete` permissions for various API endpoints.

For Twenty the example MVP could be to add selectable scopes like `read,create,update,delete` for People/Companies/Opportunities and disable of permissions for other actions.

_Screenshot for reference:_ 
<img width="910" alt="api-scopes" src="https://github.com/twentyhq/twenty/assets/4459421/4e3901a0-be38-40b4-aa13-c805fe3a1fcd">

## Technical inputs

N/A

**Comment by FelixMalfait** [comment]
Hey this is a great suggestion but I'm going to close this as we need to build permissions for the overall app first before we implement them for API Keys. And Permission is a big/ambitious project we will not yet tackle in the next 6 months (but we will at some point for sure!).
Thanks!

Edit: actually, converting this to a discussion

**Comment by FelixMalfait** [comment]
Done! You can now attach a role to an API Key
